transaction-volume-anomaly-detection/src/agent.js

```diff
@@ -0,0 +1,315 @@
+        tracker.successfulTx.txTracker[
+          tracker.successfulTx.txTracker.length - 2
+        ];
+
+      if (count * error[0] * globalSensitivity > pred[0]) {
+        localFindings.push(
+          Finding.fromObject({
+            name: "Unusually high number of successful transactions",
+            description: `Significant increase of successful transactions have been observed from ${
+              blockEvent.blockNumber - bucketBlockSize
+            } to ${blockEvent.blockNumber}`,
+            alertId: "SUCCESSFUL-TRANSACTION-VOL-INCREASE",
+            severity: FindingSeverity.Low,
+            type: FindingType.Suspicious,
+            metadata: {
+              COUNT: count,
+              EXPECTED_BASELINE: baseline,
+            },
+          })
+        );
+      }
+    }
+
+    {
+      const [pred, error] = tracker.failedTx.TSA.predict(1);
+      tracker.failedTx.TSA.predict(1);
+
+      const count =
+        tracker.failedTx.txTracker[tracker.failedTx.txTracker.length - 1];
+      const baseline =
+        tracker.failedTx.txTracker[tracker.failedTx.txTracker.length - 2];
+
+      if (count * error[0] * globalSensitivity > pred[0]) {
+        localFindings.push(
+          Finding.fromObject({
+            name: "Unusually high number of failed transactions",
+            description: `Significant increase of failed transactions have been observed from  ${
+              blockEvent.blockNumber - bucketBlockSize
+            } to ${blockEvent.blockNumber}`,
+            alertId: "FAILED-TRANSACTION-VOL-INCREASE",
+            severity: FindingSeverity.High,
+            type: FindingType.Exploit,
+            metadata: {
+              COUNT: count,
+              EXPECTED_BASELINE: baseline,
+            },
+          })
+        );
+      }
+    }
+
+    {
+      const [pred, error] = tracker.successfulInternalTx.TSA.predict(1);
+
+      const count =
+        tracker.successfulInternalTx.txTracker[
+          tracker.successfulInternalTx.txTracker.length - 1
+        ];
+      const baseline =
+        tracker.successfulInternalTx.txTracker[
+          tracker.successfulInternalTx.txTracker.length - 2
+        ];
+
+      if (count * error[0] * globalSensitivity > pred[0]) {
+        localFindings.push(
+          Finding.fromObject({
+            name: "Unusually high number of successful internal transactions",
+            description: `Significant increase of successful internal transactions have been observed from ${
+              blockEvent.blockNumber - bucketBlockSize
+            } to ${blockEvent.blockNumber}`,
+            alertId: "SUCCESSFUL-INTERNAL-TRANSACTION-VOL-INCREASE",
+            severity: FindingSeverity.Low,
+            type: FindingType.Suspicious,
+            metadata: {
+              COUNT: count,
+              EXPECTED_BASELINE: baseline,
+            },
+          })
+        );
+      }
+    }
+
+    {
+      const [pred, error] = tracker.failedInternalTx.TSA.predict(1);
```

Comment:
To get the maximum value that is not an anomaly (if the value is in the 95% confidence interval) we should use `pred[i] + 1.96 * Math.sqrt(e)`